src/FilterButtons/HideDoneBtn.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import UseTodos from '../hooks/useTodos';
+import UseFilterTodos from '../hooks/useFilterTodos';
+
+const HideDoneBtn = () => {
+
+    const { tasks } = UseTodos();
+    const { setFilterTasks } = UseFilterTodos();
+
+    const handleHideDone = () => {
+        const updatedTasks = tasks.filter(task => task.isChecked === false);
```

Comment:
find way to do it in O(1), hint: save the todo in keyBy, and than every operation of change in dict is O(1)